packages/core/src/ai-tracing/model-tracing.ts

```diff
@@ -33,10 +34,50 @@ export class ModelSpanTracker {
     this.modelSpan = modelSpan;
   }
 
+  /**
+   * Get the tracing context for creating child spans.
+   * Returns the current step span if active, otherwise the model span.
+   */
+  getTracingContext(): { currentSpan?: AnyAISpan } {
+    return {
+      currentSpan: this.currentStepSpan ?? this.modelSpan,
+    };
+  }
+
+  /**
+   * Report an error on the generation span
+   */
+  reportGenerationError(options: { error: MastraError | Error; endSpan?: boolean }): void {
+    this.modelSpan?.error(options);
+  }
+
+  /**
+   * End the generation span
+   */
+  endGeneration(options?: {
+    output?: any;
+    attributes?: Partial<ModelGenerationAttributes>;
+    metadata?: Record<string, any>;
+  }): void {
+    this.modelSpan?.end(options);
+  }
+
+  /**
+   * Update the generation span
+   */
+  updateGeneration(options: {
+    input?: any;
+    output?: any;
+    attributes?: Partial<ModelGenerationAttributes>;
+    metadata?: Record<string, any>;
+  }): void {
+    this.modelSpan?.update(options);
+  }
+
   /**
    * Start a new Model execution step
    */
-  startStepSpan(payload?: StepStartPayload) {
+  private startStepSpan(payload?: StepStartPayload) {
```

Comment:
use actual private methods
```suggestion
  #startStepSpan(payload?: StepStartPayload) {
```